### PR TITLE
Added C wrapper generator

### PR DESCRIPTION
**Pull request by stemann**
Added OCaml-based wrapper generator.

Contributes to resolution of #54 

### PR Checklist

- [x] <strike>Tests are added</strike> N/A
- [X] Documentation, if applicable

### CODE REVIEW

**Comment by stemann** [comment]
Slimmed this PR down to just adding the OCaml-based C wrapper generator - will follow up with PR updating C wrapper for Torch 1.10 (for a start) - based on the ocaml-torch wrapper generator as mentioned in https://github.com/JuliaPackaging/Yggdrasil/pull/6004#issuecomment-1356777370

File: .vscode/settings.json
Comment (ToucheSir) [inline .vscode/settings.json:1]: I think our general policy is not to have `.vscode` and similar folders checked in? Feel free to add them to the gitignore though.

Comment (stemann) [reply .vscode/settings.json:1]: Right, I assumed so too - it was just that configuration of CMake was sufficiently complex that I thought it was perhaps a case for an exception - to make it easier for a Julia user to contribute to a complex of OCaml, CMake, C etc.

Comment (stemann) [reply .vscode/settings.json:1]: … but it could be in the deps/README as well.

Comment (ToucheSir) [reply .vscode/settings.json:1]: That sounds good. It's a short enough snippet that asking users to copy-paste into their own env is perfectly reasonable.

Comment (stemann) [reply .vscode/settings.json:1]: Done

**Review by ToucheSir — APPROVED** [review]
LMK if you still have anything to add, otherwise will merge first thing tomorrow

**Comment by stemann** [comment]
> LMK if you still have anything to add, otherwise will merge first thing tomorrow

Nothing more at this point.

**Pull request merged**